server.py

```
@@ -1,103 +1,150 @@
+            self.status_code = status_code
+        self.payload = payload
+
+    def to_dict(self):
+        rv = dict(self.payload or ())
+        rv['message'] = self.message
+        return rv
+
+
+class SurveyResponse(db.Model):
+    __tablename__ = 'responses'
+    tx_id = db.Column("tx_id",
+                      UUID,
+                      primary_key=True)
+
+    ts = db.Column("ts",
+                   db.TIMESTAMP(timezone=True),
+                   server_default=db.func.now(),
+                   onupdate=db.func.now())
+
+    invalid = db.Column("invalid",
+                        db.Boolean,
+                        default=False)
+
+    data = db.Column("data", JSONB)
+
+    def __init__(self, tx_id, invalid, data):
+        self.tx_id = tx_id
+        self.invalid = invalid
+        self.data = data
+
+    def __repr__(self):
+        return '<SurveyResponse {}>'.format(self.tx_id)
+
+    def to_dict(self):
+        return {c.name: getattr(self, c.name) for c in self.__table__.columns}
+
+
 def _get_value(key):
     value = os.getenv(key)
     if not value:
         raise ValueError("No value set for " + key)
 
 
 def check_default_env_vars():
+    env_vars = ["RABBIT_CS_QUEUE", "RABBIT_CTP_QUEUE", "RABBIT_CORA_QUEUE",
+                "RABBITMQ_HOST", "RABBITMQ_PORT", "RABBITMQ_DEFAULT_USER",
+                "RABBITMQ_DEFAULT_PASS", "RABBITMQ_DEFAULT_VHOST",
+                "RABBITMQ_HOST2", "RABBITMQ_PORT2",
+                ]
 
-    env_vars = ["MONGODB_URL", "RABBIT_CS_QUEUE", "RABBIT_CTP_QUEUE", "RABBIT_CORA_QUEUE",
-                "RABBITMQ_HOST", "RABBITMQ_PORT", "RABBITMQ_DEFAULT_USER", "RABBITMQ_DEFAULT_PASS",
-                "RABBITMQ_DEFAULT_VHOST", "RABBITMQ_HOST2", "RABBITMQ_PORT2"]
-
+    missing = False
     for i in env_vars:
         try:
             _get_value(i)
         except ValueError as e:
             logger.error("Unable to start service", error=e)
-            missing_env_var = True
+            missing = True
```

Not sure I see the point of the missing var, surely it could just exit that this point as soon as it finds the first missing one?

---

I raised the same issue with @gavinleeedwards - it makes sense from a startup point of view, as you only need to fail once to understand exactly what envvars are missing.

---

Yep makes sense, I now see the point of the missing var ;)